src/app/app.component.ts

```diff
@@ -12,12 +12,15 @@ export class AppComponent implements OnInit {
 
   constructor(private readonly mainService: MainService) {}
 
-  ngOnInit() {
-    (async () => {
-      this.elements = await this.mainService.getAllElements().toPromise();
-    })();
+  ngOnInit(): void {
+     this.mainService.getAllElements().toPromise().then(
+       elements => this.elements = elements
+     ).catch(ex => {
+       // TODO: do something with error
+     });
+
   }
-  onSelected(elementUri: string) {
+  onSelected(elementUri: string): void {
```

Comment:
Followed typescript semantic to have return type so it'll be easier to review code by reviewers.